pkg/registry/registry_test.go

```diff
@@ -0,0 +1,254 @@
+func (m *mockDriver) DisplayConnection(_ context.Context) (string, error)         { return "", nil }
+func (m *mockDriver) CreateSnapshot(_ context.Context, _ string) error            { return nil }
+func (m *mockDriver) ApplySnapshot(_ context.Context, _ string) error             { return nil }
+func (m *mockDriver) DeleteSnapshot(_ context.Context, _ string) error            { return nil }
+func (m *mockDriver) ListSnapshots(_ context.Context) (string, error)             { return "", nil }
+func (m *mockDriver) Register(_ context.Context) error                            { return nil }
+func (m *mockDriver) Unregister(_ context.Context) error                          { return nil }
+func (m *mockDriver) ForwardGuestAgent() bool                                     { return false }
+func (m *mockDriver) GuestAgentConn(_ context.Context) (net.Conn, string, error)  { return nil, "", nil }
+func (m *mockDriver) Info() driver.Info                                           { return driver.Info{DriverName: m.Name} }
+func (m *mockDriver) Configure(_ *store.Instance, _ int) *driver.ConfiguredDriver { return nil }
+
+func TestRegister(t *testing.T) {
+	originalInternalDrivers := internalDrivers
+	t.Cleanup(func() {
+		internalDrivers = originalInternalDrivers
+	})
+
+	internalDrivers = make(map[string]driver.Driver)
+
+	mockDrv := newMockDriver("test-driver")
+	mockDrv2 := newMockDriver("test-driver-2")
+	Register(mockDrv)
+	Register(mockDrv2)
+
+	assert.Equal(t, len(internalDrivers), 2)
+	assert.Equal(t, internalDrivers["test-driver"], mockDrv)
+	assert.Equal(t, internalDrivers["test-driver-2"], mockDrv2)
+
+	// Test registering duplicate driver (should not overwrite)
+	mockDrv3 := newMockDriver("test-driver")
+	Register(mockDrv3)
+
+	assert.Equal(t, len(internalDrivers), 2)
+	assert.Equal(t, internalDrivers["test-driver"], mockDrv)
+
+	driverType := CheckInternalOrExternal("test-driver")
+	assert.Equal(t, driverType, Internal)
+
+	extDriver, intDriver, exists := Get("test-driver")
+	assert.Equal(t, exists, true)
+	assert.Assert(t, extDriver == nil)
+	assert.Assert(t, intDriver != nil)
+	assert.Equal(t, intDriver.Info().DriverName, "test-driver")
+
+	vmTypes := List()
+	assert.Equal(t, vmTypes["test-driver-2"], Internal)
+}
+
+func TestDiscoverDriversInDir(t *testing.T) {
+	originalExternalDrivers := ExternalDrivers
+	t.Cleanup(func() {
+		ExternalDrivers = originalExternalDrivers
+	})
+
+	ExternalDrivers = make(map[string]*ExternalDriver)
+	tempDir := t.TempDir()
+
+	var driverPath string
+	driverName := "mockext"
+	if runtime.GOOS == "windows" {
+		driverPath = filepath.Join(tempDir, "lima-driver-"+driverName+".exe")
+	} else {
+		driverPath = filepath.Join(tempDir, "lima-driver-"+driverName)
+	}
+
+	err := os.WriteFile(driverPath, []byte(""), 0o755)
+	assert.NilError(t, err)
+	t.Cleanup(func() {
+		err := os.Remove(driverPath)
+		assert.NilError(t, err)
+
+		_, err = os.Stat(driverPath)
+		assert.Assert(t, os.IsNotExist(err))
+	})
```

Comment:
This is not needed since `t.TempDir()` automatically removes a directory on Cleanup.